api/src/main/java/org/openmrs/module/fhir2/providers/r4/EncounterFhirResourceProvider.java

```diff
@@ -163,4 +170,70 @@ public IBundleProvider getEncounterEverything(@IdParam IdType encounterId) {
 		
 		return encounterService.getEncounterEverything(encounterReference);
 	}
+	
+	/**
+	 * Custom search endpoint that fetches encounters that include medication requests Returns a bundle
+	 * that includes the medication requests and any medication dispenses that reference those requests
+	 *
+	 * @param date only return encounters with an encounter date greater than or equal to this date
+	 * @param status if set to active, when determined encounters to include, exclude encounters that
+	 *            *only* have completed or cancelled medication requests
+	 * @param patientSearchTerm restrict to encounters for patients who name or identifier matches the
+	 *            search term
+	 * @return
+	 */
+	
+	@Search(queryName = "encountersWithMedicationRequests")
+	public IBundleProvider getEncountersWithMedicationRequestsSearch(@OptionalParam(name = Encounter.SP_DATE) DateParam date,
+	        @OptionalParam(name = "status") TokenParam status,
+	        @OptionalParam(name = "patientSearchTerm") TokenParam patientSearchTerm) {
+		
+		EncounterSearchParams params = new EncounterSearchParams();
+		
+		// we always want encounters, not visits
+		params.setTag(new TokenAndListParam()
+		        .addAnd(new TokenParam().setSystem(OPENMRS_FHIR_EXT_ENCOUNTER_TAG).setValue("encounter")));
+		
+		if (date != null && !date.isEmpty()) {
```

Comment:
It would definitely be good to have tests that hit at least this line (see my above comment on handling dates)